src/main/java/seedu/address/logic/commands/TagAddCommand.java

```diff
@@ -0,0 +1,211 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a tag to a contact.
+ */
+public class TagAddCommand extends Command {
+
+    public static final String COMMAND_WORD = "add";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a tag to the contact. "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_TAG + "TAG...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_TAG + "owesMoney";
+
+    public static final String MESSAGE_ADD_TAG_SUCCESS = "Tag added";
+    public static final String MESSAGE_NO_SUCH_TAG = "this tag does not exist";
```

Comment:
Should this message be capitalized like the other messages?

Reply:
I believe so. I see these messages as constants.